developer/best_practices.md

```diff
@@ -0,0 +1,188 @@
+application images.
+
+Please understand this document as a recommendation from the EDC project committers team that they compiled to the best
+of their knowledge. We realize that use case scenarios are plentiful and requirements vary, and not every best practice
+is applicable everywhere. You know your use case best.
+
+This document is not an exhaustive list of prescribed steps, that will shield adopters from any conceivable harm or
+danger, but rather should serve as starting point for engineers to build upon.
+
+Finally, it should go without saying that the software of the EDC project is distributed "as is", and committers of EDC
+take no responsibility or liability, direct or indirect, for any harm or damage caused by the use of it. This document
+does not change that.
+
+## 1. Security recommendations
+
+### 1.1 Exposing APIs to the internet
+
+The EDC code base has several outward-facing APIs, exclusively implemented as HTTP/REST endpoints. These have different
+purposes, different intended consumers and thus different security implications.
+
+As a general rule, _APIs should not be exposed directly to the internet_. That does not mean that they shouldn't be
+_accessible_ via the internet, obviously the connector and related components cannot work without a network connection.
+This only means that API endpoints should not be _directly_ facing the internet, there should
+be [appropriate infrastructure](#13-use-appropriate-network-infrastructure) in place.
+
+However, we advise __extreme caution__ when making APIs accessible via the internet - by default only the DSP API and
+the data plane's public API should be accessible via the internet, the others (management API, signaling API,...) are
+intended _only for local network access_.
+
+There are those who think that only HTTPS/TLS connections should be used, even between pods in a Kubernetes cluster.
+While the EDC project makes no argument pro or contra, that is certainly an idea worth considering in high security
+environments.
+
+The key take-away is that all of EDC's APIs - if accessible outside the local network - should *only* be accessible
+through separate software components such as API gateways or load balancers. These are specialized tools with sole
+purpose of performing authentication, authorization, rate limiting, IP blacklisting/whitelisting etc.
+
+In the particular case of the DSP API, the same principle holds, although with the exception of authentication and
+authorization. That is handled by the [DSP protocol itself]().
+
+There is a plethora of ready-made components available, both commercial and open-source, therefor the EDC project _will
+not provide that functionality._ Requests and issues to that effect will be ignored.
+
+We have a rudimentary token-based API security module available, which can be used to secure the connection API
+gateway <-> connector if so desired.
+It should be noted that it is _not designed to act a port-of-entry!_
```

Comment:
Is the term port-of-entry actually used in the context of software? Was maybe "point-of-entry" intended?